File: src/ConvertBSTToGreaterTree.java
public class ConvertBSTToGreaterTree {
    int value = 0;

    public TreeNode convertBST(TreeNode root) {
        computeGreaterTree(root);
        return root;
    }

    private void computeGreaterTree(TreeNode root) {
        if (root == null) return;
        computeGreaterTree(root.right);
        root.val += value;
        computeGreaterTree(root.left);
    }
}

File: python/convert_bst_to_greater_tree.py
# Definition for a binary tree node.
class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right


class Solution:
    def __init__(self):
        self.value = 0

    def computeGreaterTree(self, root: TreeNode):
        if root is None: return
        self.computeGreaterTree(root.right)
        root.val += self.value
        self.value = root.val
        self.computeGreaterTree(root.left)

    def convertBST(self, root: TreeNode) -> TreeNode:
        self.computeGreaterTree(root)
        return root
